applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c

```diff
@@ -1371,20 +1317,101 @@ int le_audio_send(struct encoded_audio enc_audio)
 	return 0;
 }
 
-int le_audio_enable(le_audio_receive_cb recv_cb, le_audio_timestamp_cb timestmp_cb)
+int unicast_client_enable(le_audio_receive_cb recv_cb)
 {
 	int ret;
+	static bool initialized;
+	int headset_iterator = 0;
+	int stream_iterator = 0;
+	struct bt_bap_unicast_group_stream_pair_param pair_params[ARRAY_SIZE(headsets)];
+	/* 2 streams (one sink and one source stream) for each headset */
+	struct bt_bap_unicast_group_stream_param group_stream_params[(ARRAY_SIZE(headsets) * 2)];
+	struct bt_bap_unicast_group_param group_param;
+
+	if (initialized) {
+		LOG_WRN("Already initialized");
+		return -EALREADY;
+	}
+
+	if (recv_cb == NULL) {
+		LOG_ERR("Receive callback is NULL");
+		return -EINVAL;
+	}
 
-	ret = initialize(recv_cb, timestmp_cb);
+	receive_cb = recv_cb;
+
+	LOG_DBG("Start workers");
+
+	for (int i = 0; i < ARRAY_SIZE(headsets); i++) {
+		headsets[i].sink_stream.ops = &stream_ops;
+		headsets[i].source_stream.ops = &stream_ops;
+		k_work_init_delayable(&headsets[i].stream_start_sink_work, work_stream_start);
+		k_work_init_delayable(&headsets[i].stream_start_source_work, work_stream_start);
+	}
+
+	ret = bt_bap_unicast_client_register_cb(&unicast_client_cbs);
+	if (ret != 0) {
+		LOG_ERR("Failed to register client callbacks: %d", ret);
+		return ret;
+	}
+
+	for (int i = 0; i < ARRAY_SIZE(headsets); i++) {
+		headsets[i].iso_tx_pool = iso_tx_pools[i];
+		switch (i) {
+		case AUDIO_CH_L:
+			headsets[i].ch_name = "LEFT";
+			break;
+		case AUDIO_CH_R:
+			headsets[i].ch_name = "RIGHT";
+			break;
+		default:
+			LOG_WRN("Trying to set name to undefined channel");
+			headsets[i].ch_name = "UNDEFINED";
+			break;
+		}
+	}
+
+	for (int i = 0; i < ARRAY_SIZE(group_stream_params); i++) {
+		/* Every other stream should be sink or source */
+		if ((i % 2) == 0) {
+			group_stream_params[i].qos = &lc3_preset_sink.qos;
+			group_stream_params[i].stream = &headsets[headset_iterator].sink_stream;
+		} else {
+			group_stream_params[i].qos = &lc3_preset_source.qos;
+			group_stream_params[i].stream = &headsets[headset_iterator].source_stream;
+			headset_iterator++;
+		}
+	}
+
+	for (int i = 0; i < ARRAY_SIZE(pair_params); i++) {
+		pair_params[i].tx_param = &group_stream_params[stream_iterator];
+		stream_iterator++;
+
+		if (IS_ENABLED(CONFIG_BT_AUDIO_RX)) {
```

Comment:
Seeing `CONFIG_BT_AUDIO_RX` seems strange. Is it possible to use the SINK/SOURCE checks here?

Reply:
Sinks and sources are related to the ep we discover, but BT_RX in this case is related to if we as a unicast_client have the capability to receive data as well as sending it out. We might want to add a parameter to unicast_client_enable though that specifies whether or not we would like to support both directions. That way we can reduce scope, and perhaps scheduling time, in those cases where we only want to support TX